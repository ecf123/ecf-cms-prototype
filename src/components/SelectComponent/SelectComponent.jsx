import "./SelectComponent.scss";
import topicArrow from "../../assets/upside-down-arrow-grey.svg";

const SelectComponent = () => {
  return (
     <>
     <label className="select__label">Lesson Type</label>
     <div className='select'>
         <select className='select__dropdown' src={topicArrow}>
             <option className="select__dropdownoption">Lesson</option>
             <option className="select__dropdownoption">Assessment Quiz</option>
        </select>
    </div>
    </>
  )
}
export default SelectComponent