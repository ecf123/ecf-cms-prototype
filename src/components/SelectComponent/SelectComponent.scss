@use "../../styles/_reset.scss" as *;
@use "../../styles/_variables.scss" as *;

.select {
    border: solid 1px black;
    border-radius: 0.5rem;
    width: 32rem;
    padding-left: 0.5rem;
    height: 3rem;
    display: flex;
    align-items: center;
    
    &__label{
        font-size: 0.88rem;
        color: $color-blue-dark;
        padding-bottom: 2.5rem;
    }
    
    &__dropdown {
        background: url(../../assets/upside-down-arrow-grey.svg)  no-repeat right;
        -webkit-appearance:none; 
        appearance:none;
        width: 30.63rem;
        height: 1.5rem;
        color: $color-grey-dark;
        border: none;
        font-size: 1rem;
        
        &:focus{
            outline: none;
        }
    }
}