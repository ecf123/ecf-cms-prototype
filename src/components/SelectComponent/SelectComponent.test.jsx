import { render, screen } from "@testing-library/react";
import SelectComponent from "./SelectComponent";
import { customRender } from "../../utils/testUtils";

it("Should match the snapshot", () => {
    const { container } = customRender(
      <SelectComponent />
    );
    expect(container).toMatchSnapshot();
  });

it("Should render in the SelectComponent select box", () => {
    render(<SelectComponent />)
    const input = screen.getByText("Lesson");
    expect(input).toBeInTheDocument();
});

it("Should render the SelectComponent label", () => {
    render(<SelectComponent />);
    const label = screen.getByText("Lesson Type");
    expect(label).toBeInTheDocument();
});
  