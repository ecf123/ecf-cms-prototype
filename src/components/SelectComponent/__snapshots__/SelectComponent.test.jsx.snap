// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div>
  <div>
    <label
      class="select__label"
    >
      Lesson Type
    </label>
    <div
      class="select"
    >
      <select
        class="select__dropdown"
        src="upside-down-arrow-grey.svg"
      >
        <option
          class="select__dropdownoption"
        >
          Lesson
        </option>
        <option
          class="select__dropdownoption"
        >
          Assessment Quiz
        </option>
      </select>
    </div>
  </div>
</div>
`;
